packages/lit-html/src/directives/cache.ts

```diff
@@ -21,13 +22,25 @@ import {
   clearPart,
   getCommittedValue,
   insertPart,
+  isCompiledTemplateResult,
   isTemplateResult,
   setCommittedValue,
 } from '../directive-helpers.js';
 
+/**
+ * @returns The template strings array for the provided template.
```

Comment:
I would comment the string contents are not compatible, since compiled template strings have already been prepared. They're only suitable for use as a key.

Reply:
Something like?

```

/**
 * The template strings array contents are not compatible between the two
 * template result types as the compiled template contains a prepared string,
 * only use for cache key.
 * @returns The template strings array for the provided template result.
 */
 ```